__tests__/app.test.js

```diff
@@ -123,3 +121,48 @@ describe("/api/articles/:article_id", () => {
   });
 });
 
+describe("/api/articles/:article_id/comments", () => {
+  test("GET - 200, To get comments with valid article_id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            article_id: 1,
+            votes: expect.any(Number),
+          });
+        });
+      });
+  });
+  test("GET - 200, valid article Id but comment not found", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toEqual([]);
+      });
+  });
+  test("GET - 404, valid but non-existing article Id", () => {
+    return request(app)
+      .get("/api/articles/99999/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Article not found!");
+      });
+  });
+  test("GET - 400, invalid sort query", () => {
+    return request(app)
+      .get("/api/articles/notId/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid sort query!");
+      });
```

Comment:
This test IS correctly checking for the 400 we want, but the name and message is a bit misleading (it isn't a 'sort' query, it is an invalid article ID).

Reply:
okay. i get. invalid article ID is more precise description.